### PR TITLE
tech holoparasites now talk like robots

### PR DESCRIPTION
**Pull request by ghost**
always bothered me!

### CODE REVIEW

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -91,6 +91,11 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 				possible_names += new type
 	namedatum = pick(possible_names)
 	updatetheme(pickedtheme)
+	
+/mob/living/simple_animal/hostile/guardian/get_spans()
+	if(pickedtheme == tech)
+		. = ..() | SPAN_ROBOT
+	else return
```
Comment (Armhulen) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: uuuuuuuuh give me a second here

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -91,6 +91,10 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 				possible_names += new type
 	namedatum = pick(possible_names)
 	updatetheme(pickedtheme)
+	
+/mob/living/simple_animal/hostile/guardian/get_spans()
```
Comment (MoreRobustThanYou) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Errrrr you're returning null if it's NOT tech. Do this:
```
/mob/living/simple_animal/hostile/guardian/get_spans()
	. = ..()
	if(pickedtheme == tech)
		. |= SPAN_ROBOT
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/guardian/guardian.dm (hunk 2 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm]: ah, thanks!

**Review by ghost — APPROVED** [review]

**Comment by Armhulen** [comment]
i know why this doesn't work but i don't know how i'd fix it

**Comment by coiax** [comment]
Holoparasite code is weird, and could probably use components from what I remember.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm (hunk 2 above)
Comment (optimumtact) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm]: the picked theme is never saved anywhere arm, you need to add a new var to the mob
`current_theme` or similar

and then in the `updatetheme` proc set the `current_theme` var

Then you can reference it in this var to return the right spans.

Also it needs to be `if(pickedtheme == "tech")`

Comment (optimumtact) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm]: Previously it's only been used to change the name/icon of the mob, so it's never been necessary to keep it around.

Comment (RemieRichards) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm]: or just add the span where the name/icon is set?

Comment (optimumtact) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm]: is there a list/var that say code picks up ?

Comment (RemieRichards) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm]: Hmm, before tongues I think there was, probably not anymore good point.

**Comment by jughu** [comment]
RAVIOLI RAVIOLI WHATS THE OBJECTIVE OLI

File: code/modules/mob/living/simple_animal/guardian/guardian.dm (hunk 2 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm]: OK LETS DO THIS!

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 3]
```diff
@@ -75,11 +76,10 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 	GLOB.parasites -= src
 	return ..()
 
-/mob/living/simple_animal/hostile/guardian/proc/setthemename(pickedtheme) //set the guardian's theme to something cool!
-	if(!pickedtheme)
-		pickedtheme = pick("magic", "tech", "carp")
+/mob/living/simple_animal/hostile/guardian/proc/setthemename() //set the guardian's theme to something cool!
```
Comment (SpaceManiac) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: This is still called with an argument in two places, liable to break stuff if those arguments are now being ignored.

Comment (optimumtact) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: yeah why did you change this at all? that wasn't the instruction at all.

set the current_theme in the update_theme proc, from the passed in value

**Comment by Spaceinaba** [comment]
this is nice because now miners can bring their stand out without getting lynched since they will talk differently than traitor ones

**Review by lzimann — CHANGES_REQUESTED** [review]
what oranges and spacemaniac said

**Comment by Armhulen** [comment]
i'll fix it today

**Comment by optimumtact** [comment]
you're a liar
